server/app.js

```diff
@@ -0,0 +1,65 @@
+const express = require('express');
+const bodyparser = require('body-parser');
+const db = require('./db/index.js');
+
+const app = express();
+
+app.set('port', 3004);
+
+app.use(bodyparser.json());
+
+app.use(bodyparser.urlencoded({extended: true}));
+
+//app.use(express.static(__dirname + '' ));
+
+
+app.get('/api/alsovieweditems', (req, res) => {
+
+  var queryString = 'select * from alsovieweditems';
+  var queryArgs = [];
+
+  db.query(queryString, queryArgs, function(err, results) {
+    if (err) {
+      res.json(err);
+    } else {
+      res.status(200).json(results);
+    }
+  });
+
+});
+
+app.get('/api/alsovieweditems/id/:id', (req, res) => {
+
+  var queryString = 'select * from alsovieweditems where id = ?';
```

Comment:
I see this causing problems for the integration of our services. This responds with just one item, not a list of items.

You do have a route below that responds with a list of items, but to use it you have to make a request with multiple ids. But where will those ids come from? The user is going to visit an item with one id, say 17. If we then make an API request for also-viewed items with id 17, we're going to get only a single item. But that's not what we want.

Reply:
I have implemented this route just for the sake of availability. It may not be necessary to use it for front end. I will keep this point in mind while implementing front end.